modules/doc/content/framework_development/code_standards.md

```diff
@@ -289,6 +291,47 @@ protected:
 
 [](---)
 
+
+## Code Commandments
+
+- Use references instead of pointers whenever possible
+    - i.e., this object lives for a shorter period of time than the object it needs to refer to does
+- Methods should return pointers to objects if returned objects are stored as pointers and references if returned objects are stored as references
+- When creating a new class:
+    - Include dependent header files in the *.C file whenever possible (i.e. the header uses only references or pointers in it's various declarations) and use forward declarations in the header file as needed
+    - One exception is when doing so would require end users to include multiple files to complete definitions of child objects (Errors typically show up as "incomplete class" or something similar during compilation)
+- Avoid using a global variable when possible.
+- Every destructor must be virtual.
+- All function definitions should be in *.C files.
+    - The only exceptions are for inline functions for speed and templates.
+- Thou shall not commit accidental insertion in a std::map by using brackets in a RHS operator unless he/she can prove it can't fail.
+- Thou shall use range-based loops or MooseIndex() based loops for iteration
+
+## Best practices
+
+-	All input parameters should always use range checked parameters where applicable
+-	Input parameters should always use MooseEnum instead of unsigned int
+-	mooseAssert should always be used to clearly state coding assumptions and to protect dividing by zero
+-	mooseException should be used in cases where the conditions typically used for a mooseAssert are likely to occur (e.g. inner Newton iteration that produces a negative value).
+-	mooseException should be used in favor of mooseError when checking run time conditions
+-	paramError/paramWarning should be used in favor of mooseError/mooseWarning when checking input file conditions
+-	Variable names shall be used that describe the parameter in favor of small names (e.g. activation_energy instead of Q)
+-	Utility::pow<n> shall be used in place of std::pow(x,n) if the power is constant
```

Comment:
if the power is a constant integer.  If the power is non-constant integer, use MathUtils::pow(x,n).